Assets/Scripts/PhoneCamera.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.IO;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class PhoneCamera : MonoBehaviour
+{
+
+
+    public delegate void CameraSaveEvent(string path);
+    public static CameraSaveEvent onTryAddPhoto;
+
+
+    private bool isCamAvailable;
+    private WebCamTexture backCamera;
+    private Texture defaultBackground;
+    [SerializeField] private RawImage background;
+    [SerializeField] private AspectRatioFitter fitter;
+
+
+
+
+    private void Start()
+    {
+        defaultBackground = background.texture;
+        WebCamDevice[] devices = WebCamTexture.devices;
+        if (devices.Length == 0)
+        {
+            Debug.LogError("No camera detected");
+            isCamAvailable = false;
+            return;
+        }
+        for (int i = 0; i < devices.Length; i++)
+        {
+#if !UNITY_EDITOR
+            if (!devices[i].isFrontFacing)
+#endif
+            {
+                backCamera = new WebCamTexture(devices[i].name, Screen.width, Screen.height);
+
+                isCamAvailable = true;
+            }
+        }
+        if (backCamera == null)
+        {
+            Debug.Log("No back camera were found");
```

Comment:
Ideally this would have a UI handle in the game, as we probably dont want a big black box on screen if their camera is broken, or they have denied us camera permissions

Reply:
Resolved